common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -277,8 +283,69 @@ public Object run() {
             }
 
             UNALIGNED = unaligned;
+
+            Object maybeException = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                @Override
+                public Object run() {
+                    try {
+                        // Java9 has jdk.internal.misc.Unsafe and not all methods are propergated to
+                        // sun.misc.Unsafe
+                        Class<?> internalUnsafeClass = getClassLoader(PlatformDependent0.class)
+                                .loadClass("jdk.internal.misc.Unsafe");
+                        Method method = internalUnsafeClass.getDeclaredMethod("getUnsafe");
+                        return method.invoke(null);
+                    } catch (ClassNotFoundException e) {
+                        return e;
+                    } catch (IllegalAccessException e) {
+                        return e;
+                    } catch (InvocationTargetException e) {
+                        return e;
+                    } catch (NoSuchMethodException e) {
+                        return e;
+                    } catch (SecurityException e) {
+                        return e;
+                    }
+                }
+            });
+
+            if (!(maybeException instanceof Throwable)) {
+                internalUnsafe = maybeException;
+                final Object finalInternalUnsafe = internalUnsafe;
+                maybeException = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                    @Override
+                    public Object run() {
+                        try {
+                            return finalInternalUnsafe.getClass().getDeclaredMethod(
+                                    "allocateUninitializedArray", Class.class, int.class);
+                        } catch (NoSuchMethodException e) {
+                            return e;
+                        } catch (SecurityException e) {
+                            return e;
+                        }
+                    }
+                });
+
+                if (maybeException instanceof Method) {
+                    try {
+                        Method m = (Method) maybeException;
+                        byte[] bytes = (byte[]) m.invoke(finalInternalUnsafe, byte.class, 8);
+                        assert bytes.length == 8;
+                        allocateArrayMethod = m;
+                    } catch (IllegalAccessException e) {
+                        maybeException = e;
+                    } catch (InvocationTargetException e) {
+                        maybeException = e;
+                    }
+                }
+            }
+            ALLOCATE_ARRAY_METHOD = allocateArrayMethod;
+
+            logger.debug("jdk.internal.misc.Unsafe.allocateUninitializedArray(): {}",
+                    ALLOCATE_ARRAY_METHOD != null ? "available" : "unavailable", maybeException);
```

Comment:
the way the logging format is we may not have enough `{}` if `maybeException` is not an exception. Should we only include `maybeException` if an exception occurred and have 2 log statements?